components/ModalWindow/ModalWindow.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import { StyleSheet, View } from "react-native";
+import {
+  Button,
+  Card,
+  Modal,
+  Text,
+  ThemeType,
+  useTheme,
+} from "@ui-kitten/components";
+
+type DeleteStageModalProps = {
+  modalContent: React.ReactElement;
+  buttonCancelText: string;
+  buttonOkText: string;
+  setShowModal: (value: React.SetStateAction<boolean>) => void;
+  showModal: boolean;
+  onOkPressFunction: () => void;
+};
+
+/**
+ * @component ModalWindow
+ * @description A general confirmation modal, providing options to cancel or proceed with some actions by pressing Ok button.
+ *
+ * @prop {React.ReactElement} modalContent - All content that should be displayed in white modal block
+ * @prop {string} buttonCancelText - The text of the button that cancels/hides modal
+ * @prop {string} buttonOkText  - The text of the submit button
+ * @prop {function} setShowModal - Function to toggle the modal's visibility.
+ * @prop {boolean} showModal - Controls the visibility of the modal.
+ * @prop {function} onOkPressFunction - Function that is triggered by user by pressing Ok button
+ */
+
+export const ModalWindow = ({
+  modalContent,
+  buttonCancelText,
+  buttonOkText,
+  setShowModal,
+  showModal,
+  onOkPressFunction,
+}: DeleteStageModalProps): React.ReactElement => {
+  const theme = useTheme();
+  const styles = makeStyles(theme);
+
+  const handleDelete = () => {
+    onOkPressFunction();
+    setShowModal(false);
+  };
+
+  return (
+    <Modal
+      visible={showModal}
+      backdropStyle={styles.backdrop}
```

Comment:
```suggestion
      backdropStyle={styles.backdrop}
      animationType="fade"
```

Eine Animation wäre nice